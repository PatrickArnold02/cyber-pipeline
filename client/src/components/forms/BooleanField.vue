<script setup>
import ToggleSwitch from 'primevue/toggleswitch';

// Props
const props = defineProps({
  // Field name inside of model object and errors
  field: {
    type: String,
    required: true
  },
  // Label text
  label: {
    type: String,
    required: true
  },
  // Boolean to disable editing
  disabled: {
    type: Boolean,
    default: false
  },
  // Validation errors from server
  errors: {
    type: Object,
    default() {
      return {}
    }
  }
})

// V-model of field to be edited
const model = defineModel()
</script>

<template>
  <div class="flex w-full align-items-center">
    <!-- InputText Component -->
    <ToggleSwitch
      :id="field"
      :disabled="disabled"
      :invalid="errors[field] ? true : false"
      v-model="model"
      class=""
      style="min-width: 2.5rem"
    />

    <!-- Label -->
    <label
      :for="field"
      class="ml-2"
      >{{ props.label }}</label
    >

    <!-- Error Text -->
    <small
      :id="field + '-help'"
      class="w-6 text-red-600"
      >{{ errors[field] ? errors[field][0].message : '' }}</small
    >
  </div>
</template>
