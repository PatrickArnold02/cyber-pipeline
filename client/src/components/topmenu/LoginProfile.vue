<script setup>
// Libraries
import { ref } from 'vue'
import { useRouter } from 'vue-router'
const router = useRouter()

// PrimeVue Components
import Menu from 'primevue/menu'
import Avatar from 'primevue/avatar'

// Stores
import { useTokenStore } from '@/stores/Token'
const tokenStore = useTokenStore()

// Menu Items
const items = ref([
  {
    label: 'Profile',
    icon: 'pi pi-cog',
    command: () => {
      router.push({ name: 'profile' })
    }
  },
  {
    label: 'Logout',
    icon: 'pi pi-sign-out',
    command: tokenStore.logout
  }
])

// Menu reference
const menu = ref()

/**
 * Toggle button handler
 *
 * @param {Event} event
 */
const toggle = (event) => {
  menu.value.toggle(event)
}

const visible = ref(false);
</script>

<template>
  <div class="p-menuitem">
    <!-- If no token present, show login button -->
    <div
      v-if="tokenStore.token == ''"
      class="p-menuitem-content"
    >
      <Button @click="visible = true" label="Login" icon="pi pi-sign-in"/>
    </div>

    <!-- If token present, assume user is logged in -->
    <div
      v-else
      class="p-menuitem-content"
    >
      <a
        class="p-menuitem-link"
        @click="toggle"
        aria-haspopup="true"
        aria-controls="profile_menu"
      >
        <Avatar
          icon="pi pi-user"
          shape="circle"
        />
      </a>
      <Menu
        ref="menu"
        id="profile_menu"
        :model="items"
        :popup="true"
      />
    </div>
  </div>

  <Dialog :draggable="false" v-model:visible="visible" modal header="Log-in/Sign-up" :style="{ width: '25rem' }">
      <div class="flex items-center gap-6 mb-6">
          <Button label="Log-in" @click="tokenStore.getToken()"/>
      </div>
      <div class="flex items-center gap-6 mb-6">
          <Button label="Create an account"/>
      </div>
  </Dialog>
</template>
