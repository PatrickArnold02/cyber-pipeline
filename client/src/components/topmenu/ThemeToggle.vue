<script setup>
// https://dev.to/tqbit/create-your-own-dark-mode-toggle-component-with-vue-js-1284
// https://primevue.org/theming/#switchthemes

// Libraries
import { onMounted, ref } from 'vue'
import { usePrimeVue } from 'primevue/config'
const PrimeVue = usePrimeVue()
import Logger from 'js-logger'

// Theme reference
const themeDark = ref(false)

/**
 * Determine the dark mode status on component mount.  
 * It should be true if any of the following cases:
 * - The user set a browser preference for dark color schemes
 * - The user previously toggled dark mode in this app, setting a value of 'dark' in localStorage
 * Note that it is important that if a preference for light is set in local storage, it overrides
 * the browser color scheme preference!
 */
onMounted(() => {
  // check for preference in localStorage
  switch(localStorage.getItem('user-theme'))
  {
    case 'dark': 
      themeDark.value = true;
      break;
    case 'light':
      themeDark.value = false;
      break;
    default:
      // if we fall through, check for preference in matchMedia
      // Also need to account for the lack of the matchMedia API, as not all browsers support it
      if(window.matchMedia) {
        // check for media preference for dark color schemes
        if(window.matchMedia('(prefers-color-scheme: dark)').matches) themeDark.value = true;
      }
  }  
})

/**
 * Update theme on change
 */
const updateTheme = () => {
  Logger.info('Update theme to ' + themeDark.value)
  document.documentElement.classList.toggle('app-dark')
  localStorage.setItem('user-theme', themeDark.value ? 'dark' : 'light')
}

/**
 * Toggle theme value and trigger update
 */
const toggleDarkMode = () => {
  themeDark.value = !themeDark.value
  updateTheme()
}
</script>

<template>
  <div class="p-menuitem">
    <div class="p-menuitem-content">
      <a
        class="p-menuitem-link"
        @click="toggleDarkMode()"
      >
        <span
          v-if="themeDark"
          class="pi pi-sun"
          v-tooltip.bottom="'Toggle Light Mode'"
        />
        <span
          v-else
          class="pi pi-moon"
          v-tooltip.bottom="'Toggle Dark Mode'"
        />
      </a>
    </div>
  </div>
</template>

<style scoped></style>
