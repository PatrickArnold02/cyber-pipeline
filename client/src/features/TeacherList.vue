<script setup>
// Libraries
import { ref } from 'vue'
import { storeToRefs } from 'pinia'

// PrimeVue Components
import ConfirmDialog from 'primevue/confirmdialog'
import { useConfirm } from 'primevue/useconfirm'
const confirm = useConfirm()
import { useToast } from 'primevue/usetoast'
const toast = useToast()
import { FilterMatchMode } from '@primevue/core/api'
import IconField from 'primevue/iconfield'
import InputIcon from 'primevue/inputicon'
import Button from 'primevue/button'
import InputText from 'primevue/inputtext'
import Toolbar from 'primevue/toolbar'
import Column from 'primevue/column'
import Tag from 'primevue/tag'
import MultiSelect from 'primevue/multiselect'
import Popover from 'primevue/popover'
import Message from 'primevue/message'
import Panel from 'primevue/panel'
import DataTable from 'primevue/datatable'
import Dialog from 'primevue/dialog'


// Custom Components
import DropDownField from '@/components/forms/DropDownField.vue'
import TextField from '@/components/forms/TextField.vue'
import BooleanField from '@/components/forms/BooleanField.vue'
import TextAreaField from '@/components/forms/TextAreaField.vue'
import TeacherColumn from '@/components/teacher/TeacherColumn.vue';
import ActionsColumn from '@/components/ActionsColumn.vue';
import EditTeacherDialog from '@/components/teacher/EditTeacherDialog.vue';

// Token
import { useTokenStore } from '@/stores/Token'
const tokenStore = useTokenStore()
const { is_admin } = storeToRefs(tokenStore)

// Stores
import { useTeachersStore } from '@/stores/Teachers'
const teachersStore = useTeachersStore()
import { useDistrictsStore } from '@/stores/Districts'
const districtsStore = useDistrictsStore()
import { useCohortsStore } from '@/stores/Cohorts'
const cohortsStore = useCohortsStore()
import { useCoursesStore } from '@/stores/Courses'
const coursesStore = useCoursesStore()

// Setup Stores
teachersStore.hydrate()
const { teachers } = storeToRefs(teachersStore)
districtsStore.hydrate()
const { districts } = storeToRefs(districtsStore)
const { cohorts } = storeToRefs(cohortsStore)
if (is_admin.value) {
  cohortsStore.hydrate()
  coursesStore.hydrate()
}
const { courses } = storeToRefs(coursesStore)

// Variables
const teacherDialog = ref(false) // controls opening the dialog
const teacherDialogHeader = ref('') // controls header for dialog
const loading = ref(false) // controls loading message
const message = ref('') // error message on dialog form
const teacher = ref({}) // item to be edited
const errors = ref({}) // form errors
const dt = ref() // datatable reference
const notesDialog = ref(false) // controls notes dialog
const notes = ref('') // notes for selected item

// Filters
const filters = ref({
  global: {
    value: '',
    matchMode: FilterMatchMode.CONTAINS
  },
  status: { value: null, matchMode: FilterMatchMode.IN },
  pd_status: { value: null, matchMode: FilterMatchMode.IN },
  cert_status: { value: null, matchMode: FilterMatchMode.IN },
  ms_status: { value: null, matchMode: FilterMatchMode.IN }
})

const openEditModal = () => {
  teacherDialog.value = true;
};
    
const closeModal = () => {
  teacherDialog.value = false;
};

/**
 * Click handler to edit an item in the datatable
 * @param {Teacher} aTeacher
 */
const editTeacher = (aTeacher) => {
  teacher.value = { ...aTeacher }
  teacherDialogHeader.value = 'Edit Teacher'
  openEditModal()
}

/**
 * Click handler for new button
 */
const newTeacher = () => {
  teacher.value = {
    name: '',
    email: '',
    eid: '',
    wid: '',
    districts: [],
    cohorts: [],
    courses: [],
    status: 0,
    pd_status: 0,
    cert_status: 0,
    ms_status: 0
  }
  teacherDialogHeader.value = 'New Teacher'
  teacherDialog.value = true
}

/**
 * Click handler to delete an item in the datatable
 *
 * @param {Teacher} aTeacher
 */
const deleteTeacher = (aTeacher) => {
  confirm.require({
    message: 'Are you sure you want to delete ' + aTeacher.name + '?',
    header: 'Danger Zone',
    icon: 'pi pi-exclamation-triangle',
    rejectLabel: 'Cancel',
    acceptLabel: 'Delete',
    rejectClass: 'p-button-secondary p-button-outlined',
    acceptClass: 'p-button-danger',
    accept: async () => {
      try {
        await teachersStore.delete(aTeacher.id)
        toast.add({
          severity: 'success',
          summary: 'Success',
          detail: 'Teacher Deleted!',
          life: 3000
        })
      } catch (error) {
        toast.add({
          severity: 'error',
          summary: 'Error',
          detail: error.response.data.error,
          life: 3000
        })
      }
    },
    reject: () => {
      // do nothing
    }
  })
}

/**
 * Show notes handler
 *
 * @param {String} notes notes to display
 */
const toggleNotes = (aTeacher, event) => {
  notes.value = aTeacher.notes
  notesDialog.value.toggle(event)
}

/**
 * Save button handler in edit form dialog
 */
const save = async () => {
  loading.value = true
  errors.value = {}
  message.value = ''
  teacher.value.districts = teacher.value.districts.filter((item) => item.id)
  try {
    if (teacher.value.id) {
      await teachersStore.update(teacher.value)
    } else {
      await teachersStore.new(teacher.value)
    }
    toast.add({ severity: 'success', summary: 'Success', detail: 'Teacher Updated!', life: 3000 })
    teacherDialog.value = false
  } catch (error) {
    console.log(error)
    if (error.response.data.data) {
      errors.value = error.response.data.data
      message.value = 'The server rejected this submission. Please correct errors listed below'
    } else if (error.response.data.message) {
      message.value = error.response.data.message
    } else {
      message.value =
        'The server rejected this submission due to an SQL Error. Refresh and try again'
    }
  }
  loading.value = false
}

/**
 * Export datatable to CSV
 */
const exportCSV = () => {
  dt.value.exportCSV()
}

/**
 * Statuses
 */
const statuses = [
  { label: 'New', id: 0, severity: 'warning', icon: 'pi pi-star', hidden: 'false' },
  { label: 'Active', id: 1, severity: 'primary', icon: 'pi pi-sync', hidden: 'false' },
  { label: 'Inactive', id: 2, severity: 'secondary', icon: 'pi pi-times', hidden: 'true' },
  { label: 'Complete', id: 3, severity: 'success', icon: 'pi pi-check', hidden: 'false' }
]

const grades = [
  { label: 'Enrolled', id: 0, severity: 'primary', icon: 'pi pi-circle' },
  { label: 'Pass', id: 1, severity: 'success', icon: 'pi pi-check' },
  { label: 'Incomplete', id: 2, severity: 'warning', icon: 'pi pi-info' },
  { label: 'Fail', id: 3, severity: 'danger', icon: 'pi pi-times' },
  { label: 'Withdrawn', id: 4, severity: 'secondary', icon: 'pi pi-minus' }
]

/**
 * Custom export function to handle exporting datatable data
 *
 * @param {Teacher} row
 */
const exportFunction = (row) => {
  if (row.field == 'districts') {
    return row.data.find((item) => item.primary == true)?.usdName || ''
  } else if (row.field == 'cohorts') {
    return row.data.map((item) => item.name).join(' ')
  } else if (row.field == 'courses') {
    return row.data
      .map((item) => item.name + ': ' + grades.find((grade) => grade.id == item.status).label)
      .join('","')
  } else if (row.field.endsWith('status')) {
    return statuses.find((status) => status.id == row.data)?.label || ''
  } else {
    return row.data
  }
}
</script>

<template>
  <!-- Location for confirmation dialog to be inserted -->
  <ConfirmDialog></ConfirmDialog>

  <!-- Main datatable for items -->
  <Panel header="Manage Teachers">
    <DataTable
      ref="dt"
      :value="teachers"
      stripedRows
      sortField="name"
      :sortOrder="1"
      tableStyle="min-width: 50rem"
      v-model:filters="filters"
      filterDisplay="row"
      :globalFilterFields="['name', 'email', 'eid', 'wid', 'all_districts', 'grade_level']"
      :exportFunction="exportFunction"
    >
      <template #header>
        <Toolbar
          class="mb-2"
          style="border: none"
        >
          <template #start>
            <Button
              v-if="is_admin"
              label="New"
              icon="pi pi-plus"
              severity="success"
              class="mr-2"
              @click="newTeacher"
            />
            <Button
              label="Export"
              icon="pi pi-upload"
              severity="help"
              @click="exportCSV($event)"
            />
          </template>
          <template #end>
            <div class="flex justify-content-end">
              <IconField iconPosition="left">
                <InputIcon>
                  <i class="pi pi-search" />
                </InputIcon>
                <InputText
                  v-model="filters['global'].value"
                  placeholder="Keyword Search"
                />
              </IconField>
            </div>
          </template>
        </Toolbar>
      </template>
      <template #empty>
        <div class="p-text-center">
          <p>No Teachers Found</p>
        </div>
      </template>
      <Column
        field="name"
        sortable
        header="Name"
      ></Column>
      <Column
        field="email"
        sortable
        header="Email"
      ></Column>
      <Column
        field="eid"
        sortable
        header="eID"
      ></Column>
      <Column
        field="wid"
        sortable
        header="WID"
      ></Column>
      <TeacherColumn
        v-if="is_admin"
        :header="'Status'"
        :field="'status'"
        :statuses="statuses"
      />
      <TeacherColumn
        v-if="is_admin"
        :header="'PD'"
        :field="'pd_status'"
        :statuses="statuses"
      />
      <TeacherColumn
        v-if="is_admin"
        :header="'Cert'"
        :field="'cert_status'"
        :statuses="statuses"
      />
      <TeacherColumn
        v-if="is_admin"
        :header="'MS'"
        :field="'ms_status'"
        :statuses="statuses"
      />
      <Column
        field="grade_level"
        header="Grade Level"
        sortable
      ></Column>
      <Column
        field="districts"
        header="Districts"
      >
        <template #body="slotProps">
          <Tag
            v-for="district in slotProps.data.districts"
            :key="district.id"
            :value="district.usdName"
            :severity="district.primary == true ? 'success' : 'secondary'"
            class="m-1"
          />
        </template>
      </Column>
      <Column
        field="cohorts"
        header="Cohorts"
      >
        <template #body="slotProps">
          <Tag
            v-for="cohort in slotProps.data.cohorts"
            :key="cohort.id"
            :value="cohort.name"
            class="m-1"
          />
        </template>
      </Column>
      <Column
        v-if="is_admin"
        field="courses"
        header="Courses"
      >
        <template #body="slotProps">
          <Tag
            v-for="course in slotProps.data.courses"
            :key="course.id"
            :value="course.name"
            :severity="grades[course.status].severity"
            :icon="grades[course.status].icon"
            class="m-1"
          />
        </template>
      </Column>
      <Column
        header="Actions"
        :exportable="false"
        style="min-width: 8rem"
      >
        <template #body="slotProps">
          <Button
            v-if="is_admin"
            icon="pi pi-pencil"
            outlined
            rounded
            class="mr-2"
            @click="editTeacher(slotProps.data)"
            v-tooltip.bottom="'Edit'"
          />
          <Button
            v-if="is_admin"
            icon="pi pi-trash"
            outlined
            rounded
            class="mr-2"
            severity="danger"
            @click="deleteTeacher(slotProps.data)"
            v-tooltip.bottom="'Delete'"
          />
          <Button
            v-if="slotProps.data.notes && slotProps.data.notes.length > 0"
            icon="pi pi-file"
            outlined
            rounded
            severity="info"
            @click="toggleNotes(slotProps.data, $event)"
            v-tooltip.bottom="'Notes'"
          />
        </template>
      </Column>
      <ActionsColumn/>
    </DataTable>
  </Panel>

  <!-- Notes dialog -->
  <Popover ref="notesDialog">
    <div class="flex flex-column gap-1 w-25rem">
      <div class="w-full">
        <span>Notes</span>
        <hr class="w-full" />
      </div>
      <span>{{ notes }}</span>
    </div>
  </Popover>

  <!-- Edit item dialog -->
  <EditTeacherDialog
    v-if="is_admin"
    :teacher-dialog="teacherDialog"
    :teacher-dialog-header="teacherDialogHeader"
    :teacher="teacher"
    :errors="errors"
    :statuses="statuses"
    @close-modal="closeModal"
  />
</template>

<style scoped>
:deep(.p-datatable-header) {
  padding: 0px !important;
}
</style>
