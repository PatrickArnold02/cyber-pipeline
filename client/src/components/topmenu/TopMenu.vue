<script setup>
// Libraries
import { ref } from 'vue'
import { useRouter } from 'vue-router'
const router = useRouter()

// PrimeVue Components
import Menubar from 'primevue/menubar'


// Custom Components
import ThemeToggle from '@/components/topmenu/ThemeToggle.vue'
import LoginProfile from '@/components/topmenu/LoginProfile.vue'
import { useToast } from 'primevue/usetoast'
const toast = useToast()

// Stores
import { useTokenStore } from '@/stores/Token'
const tokenStore = useTokenStore()

import { useCohortsStore } from '@/stores/Cohorts.js'
import { useCoursesStore } from '@/stores/Courses.js'
import { useUsersStore } from '@/stores/Users.js'
import { useDistrictsStore } from '@/stores/Districts.js'
import { useTeachersStore } from '@/stores/Teachers.js'
import { useEnrollmentStore } from '@/stores/Enrollment.js'
import { useRolesStore } from '@/stores/Roles.js'
import { useCanvasStore } from '@/stores/Canvas.js'

const cohortsStore = useCohortsStore()
const coursesStore = useCoursesStore()
const usersStore = useUsersStore()
const districtsStore = useDistrictsStore()
const teacherStore = useTeachersStore()
const enrollmentStore = useEnrollmentStore()
const rolesStore = useRolesStore()
const canvasStore = useCanvasStore()

const testCanvasAPI = async () => {
  toast.add({
    severity: 'info',
    summary: 'Testing API Connection',
    detail: 'Trying to pull all courses from the Canvas API',
    life: 3000
  })

  const response = await canvasStore.getCourses()

  if(response.status != 200){
    toast.add({
      severity: 'error',
      summary: 'Error',
      detail: 'Unable to pull all courses from the Canvas API',
      life: 3000
    })
  } else if(response.status == 200) {
    toast.add({
      severity: 'success',
      summary: 'API Connection Successful',
      detail: 'Successfully pulled all courses from the Canvas API',
      life: 3000
    })
  }
}

/**
 * Rehydrates all the stores, slightly faster than fully refreshing the page
 */
const syncWithDatabase = async () => {
  try{
    toast.add({
      severity: 'info',
      summary: 'Syncing with Database',
      detail: 'Please wait...',
      life: 3000
    })

    await cohortsStore.hydrate()
    await coursesStore.hydrate()
    await usersStore.hydrate()
    await districtsStore.hydrate()
    await teacherStore.hydrate()
    await enrollmentStore.hydrate()
    await rolesStore.hydrate()

    toast.add({
      severity: 'success',
      summary: 'Synced with Database',
      detail: 'All stores have been hydrated',
      life: 3000
    })
  } catch(error){
    toast.add({
      severity: 'error',
      summary: 'Error',
      detail: 'Unable to sync with database',
      life: 3000
    })
  }  
}

// Menu items
const items = ref([
  {
    label: 'Home',
    icon: 'pi pi-home',
    command: () => {
      router.push({ name: 'home' })
    }
  }
])

/**
 * Watch for changes in token and update menu
 */
tokenStore.$subscribe(() => {
  // All users render these fields
  if (tokenStore.is_user || tokenStore.is_admin) {
    items.value = [
      {
        label: 'Home',
        icon: 'pi pi-home',
        command: () => {
          router.push({ name: 'home' })
        }
      },
      {
        label: 'Teachers',
        icon: 'pi pi-users',
        command: () => {
          router.push({ name: 'teachers' })
        }
      },
      {
        label: 'Districts',
        icon: 'pi pi-building',
        command: () => {
          router.push({ name: 'districts' })
        }
      }
    ]
  }
  if (tokenStore.is_admin) {
    items.value.push(
      {
        label: 'Cohorts',
        icon: 'pi pi-users',
        command: () => {
          router.push({ name: 'cohorts' })
        }
      },
      {
        label: 'Courses',
        icon: 'pi pi-book',
        command: () => {
          router.push({ name: 'courses' })
        }
      },
      {
        label: 'Users',
        icon: 'pi pi-user-edit',
        command: () => {
          router.push({ name: 'users' })
        }
      },
      {
        label: 'Mailing',
        icon: 'pi pi-envelope',
        command: () => {
          router.push({ name: 'mailing'})
        }
      },
      {
        label: 'Analytics',
        icon: 'pi pi-sliders-h',
        command: () => {
          router.push({ name: 'analytics' })  
        }
      },
      {
        label: 'Sync',
        icon: 'pi pi-refresh',
        command: () => {
          syncWithDatabase()
        }
      },
      {
        label: 'Test API Connection',
        icon: 'pi pi-cloud',
        command: () => {
          testCanvasAPI()
        }
      }
    )
  }
})
</script>

<template>
  <Menubar :model="items">
    <template #start>
      <RouterLink :to="{ name: 'home' }">
        <img
          src="../../assets/logo.png"
          height="46px"
          alt="CyberPipeline Logo"
        />
      </RouterLink>
    </template>
    <template #end>
      <div class="flex align-items-center gap-2">
        <ThemeToggle /> 
        <LoginProfile />
      </div>
    </template>
  </Menubar>
</template>

<style scoped></style>
