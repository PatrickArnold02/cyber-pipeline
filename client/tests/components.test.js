import { mount, shallowMount } from '@vue/test-utils'
import { createPinia, setActivePinia } from 'pinia'
import { describe, it, beforeEach, expect } from 'vitest'
import TopMenu from '@/components/topmenu/TopMenu.vue'
import ThemeToggle from '@/components/topmenu/ThemeToggle.vue'
import LoginProfile from '@/components/topmenu/LoginProfile.vue'
import UserList from '@/components/user/UserList.vue'
import TeacherDashboard from '@/components/teacher/TeacherDashboard.vue'
import TeacherList from '@/components/teacher/TeacherList.vue'
import ProfileEdit from '@/components/profile/ProfileEdit.vue'
import AutocompleteMultiple from '@/components/forms/AutocompleteMultiple.vue'
import BooleanField from '@/components/forms/BooleanField.vue'
import DropDownField from '@/components/forms/DropDownField.vue'
import TextAreaField from '@/components/forms/TextAreaField.vue'
import TextField from '@/components/forms/TextField.vue'
import DistrictList from '@/components/district/DistrictList.vue'
import CourseList from '@/components/course/CourseList.vue'
import CohortDashboard from '@/components/cohort/CohortDashboard.vue'
import CohortList from '@/components/cohort/CohortList.vue'

// describe('TopMenu', () => {

//   const pinia = createPinia()
//   setActivePinia(pinia) // Makes Pinia active

//   it('should render the component correctly', () => {

//     // Mount the component with Pinia
//     const wrapper = mount(TopMenu, {
//     })

//     // Assert the component is rendered correctly
//     expect(wrapper).toBeTruthy()
//   })
// })

// describe('ThemeToggle', () => {

//   const pinia = createPinia()
//   setActivePinia(pinia) // Makes Pinia active

//   it('should render the component correctly', () => {
//     // Mount the component with Pinia
//     const wrapper = mount(ThemeToggle, {
//     })

//     // Assert the component is rendered correctly
//     expect(wrapper).toBeTruthy()
//   })
// })

describe('LoginProfile', () => {

  const pinia = createPinia()
  setActivePinia(pinia) // Makes Pinia active

  it('should render the component correctly', () => {
    // Mount the component with Pinia
    const wrapper = mount(LoginProfile, {
    })

    // Assert the component is rendered correctly
    expect(wrapper).toBeTruthy()
  })
})

describe('UserList', () => {

  const pinia = createPinia()
  setActivePinia(pinia) // Makes Pinia active

  it('should render the component correctly', () => {
    // Mount the component with Pinia
    const wrapper = mount(UserList, {
    })

    // Assert the component is rendered correctly
    expect(wrapper).toBeTruthy()
  })
})

describe('TeacherDashboard', () => {

  const pinia = createPinia()
  setActivePinia(pinia) // Makes Pinia active

  it('should render the component correctly', () => {
    // Mount the component with Pinia
    const wrapper = mount(TeacherDashboard, {
    })

    // Assert the component is rendered correctly
    expect(wrapper).toBeTruthy()
  })
})

describe('TeacherList', () => {

  const pinia = createPinia()
  setActivePinia(pinia) // Makes Pinia active

  it('should render the component correctly', () => {
    // Mount the component with Pinia
    const wrapper = mount(TeacherList, {
    })

    // Assert the component is rendered correctly
    expect(wrapper).toBeTruthy()
  })
})

describe('ProfileEdit', () => {

  const pinia = createPinia()
  setActivePinia(pinia) // Makes Pinia active

  it('should render the component correctly', () => {
    // Mount the component with Pinia
    const wrapper = mount(ProfileEdit, {
    })

    // Assert the component is rendered correctly
    expect(wrapper).toBeTruthy()
  })
})

describe('AutoCompleteMultiple', () => {

  const pinia = createPinia()
  setActivePinia(pinia) // Makes Pinia active

  it('should render the component correctly', () => {
    // Mount the component with Pinia
    const wrapper = mount(AutocompleteMultiple, {
    })

    // Assert the component is rendered correctly
    expect(wrapper).toBeTruthy()
  })
})

describe('BooleanField', () => {

  const pinia = createPinia()
  setActivePinia(pinia) // Makes Pinia active

  it('should render the component correctly', () => {
    // Mount the component with Pinia
    const wrapper = mount(BooleanField, {
    })

    // Assert the component is rendered correctly
    expect(wrapper).toBeTruthy()
  })
})

describe('DropDownField', () => {

  const pinia = createPinia()
  setActivePinia(pinia) // Makes Pinia active

  it('should render the component correctly', () => {
    // Mount the component with Pinia
    const wrapper = mount(DropDownField, {
    })

    // Assert the component is rendered correctly
    expect(wrapper).toBeTruthy()
  })
})

describe('TextAreaField', () => {

  const pinia = createPinia()
  setActivePinia(pinia) // Makes Pinia active

  it('should render the component correctly', () => {
    // Mount the component with Pinia
    const wrapper = mount(TextAreaField, {
    })

    // Assert the component is rendered correctly
    expect(wrapper).toBeTruthy()
  })
})

describe('TextField', () => {

  const pinia = createPinia()
  setActivePinia(pinia) // Makes Pinia active

  it('should render the component correctly', () => {
    // Mount the component with Pinia
    const wrapper = mount(TextField, {
    })

    // Assert the component is rendered correctly
    expect(wrapper).toBeTruthy()
  })
})

describe('DistrictList', () => {

  const pinia = createPinia()
  setActivePinia(pinia) // Makes Pinia active

  it('should render the component correctly', () => {
    // Mount the component with Pinia
    const wrapper = mount(DistrictList, {
    })

    // Assert the component is rendered correctly
    expect(wrapper).toBeTruthy()
  })
})

describe('CourseList', () => {

  const pinia = createPinia()
  setActivePinia(pinia) // Makes Pinia active

  it('should render the component correctly', () => {
    // Mount the component with Pinia
    const wrapper = mount(CourseList, {
    })

    // Assert the component is rendered correctly
    expect(wrapper).toBeTruthy()
  })
})

describe('CohortDashboard', () => {

  const pinia = createPinia()
  setActivePinia(pinia) // Makes Pinia active

  it('should render the component correctly', () => {
    // Mount the component with Pinia
    const wrapper = mount(CohortDashboard, {
    })

    // Assert the component is rendered correctly
    expect(wrapper).toBeTruthy()
  })
})

describe('CohortList', () => {

  const pinia = createPinia()
  setActivePinia(pinia) // Makes Pinia active

  it('should render the component correctly', () => {
    // Mount the component with Pinia
    const wrapper = mount(CohortList, {
    })

    // Assert the component is rendered correctly
    expect(wrapper).toBeTruthy()
  })
})







