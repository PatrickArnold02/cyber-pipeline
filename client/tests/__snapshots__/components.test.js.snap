// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`LoginProfile > should render the component correctly 1`] = `
"<div class="p-menuitem">
  <!-- If no token present, show login button -->
  <div class="p-menuitem-content"><a class="p-menuitem-link"><span class="p-menuitem-icon pi pi-sign-in"></span><span class="p-menuitem-text">Login</span></a></div>
</div>"
`;

exports[`MyComponent > renders the component correctly 1`] = `"<menubar-stub model="[object Object]"></menubar-stub>"`;
